index.php

```diff
@@ -1,34 +1,64 @@
 <?php
 
 // twig and PHPmailer composer:
-require_once 'vendor/autoload.php';
+require_once __DIR__ . '/vendor/autoload.php';
 
 $loader = new \Twig\Loader\FilesystemLoader(__DIR__ . '/view');
 $twig = new \Twig\Environment($loader, [
     'cache' => false // __DIR__ . '/tmp',
 ]);
 
 //controller :
-require('controller/controller.php');
+require __DIR__ . '/controller/controller.php';
 
 //Routing
 if (isset($_GET['action'])) {
+    $action = htmlspecialchars($_GET['action']);
     switch ($_GET['action']) {
         case 'homePage':
-            homePage($twig);
+            homePage($twig, $action);
             break;
 
         case 'contactMe':
             contactMe();
             break;
 
+        case 'blog':
+
+            // NOUVELLE VERSION
+            $postsPager=postsPager(); 
+            $posts = $postsPager['posts'];
+            $p_pages_nbr = $postsPager['p_pages_nbr'];
+
+            if (isset($_GET['page']) && $_GET['page'] > 0 && $_GET['page'] <= $p_pages_nbr) {
+                $page=(int) htmlspecialchars($_GET['page']); // htmlspecialchars nécessaire vu toutes les précautions du if ? Pense pas...
+            }
+            else {
+                $page=1;
+                header('Location: index.php?action=blog&page=1');
+            }
+
+ /*         //Why is_int() doesn't work ?   
+            if (isset($_GET['page']) && is_int($_GET['page'])===true && $_GET['page'] > 0 && $_GET['page'] <= $p_pages_nbr) {
```

Comment:
The `===true` part of your condition with `is_int` isn't needed. you can write it like `isset`.
As for why it's not working you should check the second note on this page: [https://www.php.net/manual/fr/reserved.variables.get](https://www.php.net/manual/fr/reserved.variables.get)

Reply:
I'm sorry : I think I removed this for my code. It's an oversight. I would like to discuss this matter with you in our next weekly session. Thanks for the link. I will study that, so.